deployment/stacks/api.ts

```diff
@@ -433,34 +434,39 @@ export function createApiComponents(props: CreateApiComponentsProps) {
     ...commonProps,
   });
 
-  const waf = new WafConstruct(
-    scope,
-    "ApiWafConstruct",
-    {
-      name: `${project}-${stage}-${shortStackName}`,
-      blockRequestBodyOver8KB: false,
-    },
-    "REGIONAL"
-  );
-  new wafv2.CfnWebACLAssociation(scope, "WebACLAssociation", {
-    resourceArn: api.deploymentStage.stageArn,
-    webAclArn: waf.webAcl.attrArn,
-  });
-
-  if (!isDev) {
-    const logBucket = new s3.Bucket(scope, "WafLogBucket", {
-      encryption: s3.BucketEncryption.S3_MANAGED,
-      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
-      removalPolicy: RemovalPolicy.RETAIN,
-      enforceSSL: true,
+  if (!isLocalStack()) {
+    const waf = new WafConstruct(
+      scope,
+      "ApiWafConstruct",
+      {
+        name: `${project}-${stage}-${shortStackName}`,
+        blockRequestBodyOver8KB: false,
+      },
+      "REGIONAL"
+    );
+
+    new wafv2.CfnWebACLAssociation(scope, "WebACLAssociation", {
+      resourceArn: api.deploymentStage.stageArn,
+      webAclArn: waf.webAcl.attrArn,
     });
 
-    new CloudWatchToS3(scope, "CloudWatchToS3Construct", {
-      logGroup: waf.logGroup,
-      bucket: logBucket,
-      iamPermissionsBoundary: props.iamPermissionsBoundary,
-      iamPath: props.iamPath,
-    });
+    if (!isDev) {
+      const logBucket = new s3.Bucket(scope, "WafLogBucket", {
+        encryption: s3.BucketEncryption.S3_MANAGED,
+        blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+        removalPolicy: RemovalPolicy.RETAIN,
+        enforceSSL: true,
+      });
+
+      if (waf.logGroup) {
+        new CloudWatchToS3(scope, "CloudWatchToS3Construct", {
+          logGroup: waf.logGroup,
+          bucket: logBucket,
+          iamPermissionsBoundary: props.iamPermissionsBoundary,
+          iamPath: props.iamPath,
+        });
+      }
```

Comment:
```suggestion
      new CloudWatchToS3(scope, "CloudWatchToS3Construct", {
        logGroup: waf.logGroup,
        bucket: logBucket,
        iamPermissionsBoundary: props.iamPermissionsBoundary,
        iamPath: props.iamPath,
      });
```
Shouldn't need this if anymore iirc